tests/components/evohome/conftest.py

```diff
@@ -150,6 +157,21 @@ async def setup_evohome(
 
         assert isinstance(mock_client.call_args.kwargs["session"], ClientSession)
 
-        assert mock_client.account_info is not None
+        assert evo and evo.account_info is not None
 
+        mock_client.return_value = evo
         yield mock_client
+
+
+@pytest.fixture
+async def evohome(
+    hass: HomeAssistant,
+    config: dict[str, str],
+    install: str,
+) -> AsyncGenerator[EvohomeClient]:
+    """Return the evohome client for this install fixture."""
+
+    async for mock_client in setup_evohome(hass, config, install=install):
+        evo: EvohomeClient = mock_client.return_value
+
+        yield evo
```

Comment:
I've had to implement this fixture instead of the `dhw_id` fixture.

I will convert the other tests from using `setup_evohome()`, to this fixture in a future PR.